File: Arrays/ContainsCommonItems.java
/**
 * ContainsCommonItems
 */
public class ContainsCommonItems {
    public boolean CheckCommon(String[] arr1, String[] arr2){
        for (int i = 0; i < arr1.length; i++) {
            for (int j = 0; j < arr2.length; j++) {
                if (arr1[i]==arr2[j]) {
                    return true;
                }
            }
        }
        return false;
    }

    public static void main(String[] args) {
        String[] arr1 =  new String[4];
        String[] arr2 =  new String[3];

        arr1[0] = "a";
        arr1[1] = "b";
        arr1[2] = "c";
        arr1[3] = "x";

        
        arr2[0] = "z";
        arr2[1] = "y";
        arr2[2] = "x";

        ContainsCommonItems obj = new ContainsCommonItems();
        boolean result = obj.CheckCommon(arr1, arr2);

        System.out.println(result);
    }
}

File: Arrays/ContainsCommonItems.js
// function ContainsCommonItem(arr1, arr2) {
//     for (let i = 0; i < arr1.length; i++) {
//         for (let j = 0; j < arr2.length; j++) {
//             if (arr1[i]==arr2[j]) {
//                 return true;
//             }
//         }
//     }
//     return false;
// }

// result = ContainsCommonItem(['a','b','c','x'],['z','y','i'])
// console.log(result);

// // O(a*b)

function ContainsCommonItem(arr1, arr2){
    let map = {}; 
    for (let index = 0; index < arr1.length; index++) {
        if (!map[arr1[index]]) {
            const item = arr1[index];
            map[item] = true;
        }
    }  

    for (let index = 0; index < arr2.length; index++) {
        if (map[arr2[index]]) {
            return true;
        }
    }

    return false;
}

result = ContainsCommonItem(['a','b','c','x'],['z','y','x'])
console.log(result);

// O(a+b)